tests/Unit/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/Test_GaugeWave.cpp

```diff
@@ -0,0 +1,296 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <algorithm>
+#include <cmath>
+#include <cstddef>
+#include <string>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/EagerMath/DeterminantAndInverse.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Options/Options.hpp"
+#include "Options/ParseOptions.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/GeneralRelativity/GaugeWave.hpp"
+#include "PointwiseFunctions/GeneralRelativity/ComputeSpacetimeQuantities.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "Utilities/MakeWithValue.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+#include "tests/Unit/Pypp/CheckWithRandomValues.hpp"
+#include "tests/Unit/Pypp/SetupLocalPythonEnvironment.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+
+namespace Tags {
+template <typename Tag>
+struct dt;
+}  // namespace Tags
+
+namespace {
+struct GaugeWaveProxy : gr::Solutions::GaugeWave {
+  using gr::Solutions::GaugeWave::GaugeWave;
+
+  template <typename DataType>
+  using variables_tags = gr::Solutions::GaugeWave::tags<DataType>;
+
+  template <typename DataType>
+  tuples::tagged_tuple_from_typelist<variables_tags<DataType>> test_variables(
+      const tnsr::I<DataType, 3>& x, const double t) const noexcept {
+    return variables(x, t, variables_tags<DataType>{});
+  }
+};
+
+struct GaugeWaveOptionTag {
+  using type = gr::Solutions::GaugeWave;
+  static constexpr OptionString help{"A gauge-wave solution"};
+};
+
+template <typename DataType>
+tnsr::I<DataType, 3, Frame::Inertial> spatial_coords(
+    const DataType& used_for_size) noexcept {
+  auto x = make_with_value<tnsr::I<DataType, 3, Frame::Inertial>>(used_for_size,
+                                                                  0.0);
+  get<0>(x) = 1.32;
+  get<1>(x) = 0.82;
+  get<2>(x) = 1.24;
+  return x;
+}
+
+template <typename DataType>
+void test_gauge_wave(const DataType& used_for_size) noexcept {
```

Comment:
It would be good to take the `gr::Solution::GaugeWave` as an argument, requiring it be constructed with the correct amplitude and wavelength. A good way to check that is:
```cpp
REQUIRE(solution == gr::Solutions::GaugeWave{0.24, 4.4});
```
The `REQUIRE` will not go through the rest of the test if it fails. This is desirable since we know if that check doesn't pass the rest of this function will fail.

By taking the soln as an argument you can then do things like `test_gauge_wave(serialize_and_deserialize(solution), used_for_size);` in the serialization test (I'll comment below). We've started moving toward redoing the checks for serialization and factory construction because we've sometimes had bugs in `operator==` sneak in and then it turned out serialization was broken.